net/http/CMakeLists.txt

```diff
@@ -35,6 +35,11 @@ endif()
 
 if(ssl)
   include_directories(${OPENSSL_INCLUDE_DIR})
+  add_definitions(-DNO_SSL_DL)
+  if(OPENSSL_VERSION AND (${OPENSSL_VERSION} VERSION_EQUAL "1.1"))
+     MESSAGE( STATUS "Use SSL API VERSION 1.1 for civetweb" )
+     add_definitions(-DOPENSSL_API_1_1)
```

Comment:
Why add this to CMake instead of using `OPENSSL_VERSION_NUMBER` directly in the code, as suggested in https://wiki.openssl.org/index.php/OpenSSL_1.1.0_Changes?